R/probabilities.R

```diff
@@ -1,17 +1,46 @@
 #' Probabilities
-#' \code{Probabilities} A generic function used to extract one or more
-#' variables containing probabilities relating to cases (e.g., segment membership).
-#' @param object An object for which probabilities are desired.
-#' @param ... Additional argument
+#'
+#' Estimates the probability of group membership for the data passed into \code{newdata} or
+#' the data used to fit the model if \code{newdata} is not specified. Intended to be used
+#' for the classifiers in the packages \code{flipRegression}, \code{flipMultivariates} and
+#' \code{flipTrees}.
+#'
+#' @param object A \code{CART}, \code{MachineLearning} or \code{Regression} object.
+#' @param newdata Optionally, a data frame including the variables used to fit the model.
+#'        If not provided, the object$model is used instead.
+#' @param ... Optional arguments to pass to \code{predict} or other functions.
+#' @return A matrix of predicted probabilities for the observation to belong to each
+#'         class label.
 #' @export
-Probabilities <- function(object, ...) {
+Probabilities <- function(object, newdata = NULL, ...)
+{
+    validateProbabilityArguments(object, newdata)
     UseMethod("Probabilities")
 }
 
-#' #' @inheritParams Probabilities
-#' #' @describeIn Probabilities Error occurs as no method has been specified.
-#' #' @export
-#' Probabilities.default = function(object, ...)
-#' {
-#'     stop("No 'Probabilities' method exists for this class of objects.")
-#' }
+Probabilities.default <- function(object, newdata = NULL, ...)
+{
+    stop("object not supported")
+}
+
+throwErrorUnsupportedPredictionClass <- function(valid.classes)
+{
+    n.valid <- length(valid.classes)
+    valid.classes <- sQuote(valid.classes)
+    valid.classes <- paste0(paste0(valid.classes[seq_len(n.valid - 1)], collapse = ", "),
+                            " or ", valid.classes[n.valid])
+    stop(sQuote("object"), " must be a ", valid.classes, " object. ")
+}
+
+validateProbabilityArguments <- function(object, newdata)
+{
+    valid.classes <- c("CART", "MachineLearning", "Regression")
+    not.valid.object <- !inherits(object, valid.classes)
```

Comment:
Isnt the vector support for inherits only in 4.3?

Reply:
I misread the release notes, it has been supported earlier but now has other support. I'll double check the release notes.